### ISSUE TITLE
[Bug]: Channel download fails to register all videos?

### ISSUE THREAD

**Issue by x775** [tubearchivist/tubearchivist#359]
### Latest and Greatest

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).

### Operating System

Linux

### Your Bug Report

Hi there ---

I am experiencing issues actually downloading complete channels.

One example is adding https://www.youtube.com/channel/UC3KM_fnTj9Qz2nEpZ-pB21Q, scanning all channel pages (the number seems to be correct), and otherwise proceeded to download videos. I let the system run overnight, but checking back today, I only see 58 videos (all of which are from 2022) downloading. Rescanning my subscriptions does not capture more videos despite the channel having more than 1500 videos in total. 

Unfortunately, this is not the only instance; examples vary from Business Insider (https://www.youtube.com/channel/UCcyq283he07B7_KUX07mmtA) to Marques Brownlee (https://www.youtube.com/channel/UCBJycsmduvYEL83R_U4JriQ) to Yurara Sarara (https://www.youtube.com/channel/UCjFxPoj7j3s8gECyZSM3Qjg). The only channels that appear to be complete are those whose total does not seem to exceed ~50.

I cannot immediately see anything in the logs, but am happy to provide whatever might be helpful.

Is this a known issue, or am I doing something wrong?

### Relevant log output

```shell
None
```

**Comment by lamusmaser** [comment]
This is working as designed, but this particular issue has been brought up a few times.

From the Wiki's FAQ: https://github.com/tubearchivist/tubearchivist/wiki/FAQ#5-when-i-subscribe-to-a-channel-it-only-downloads-the-most-recent-50-videos

As a quick fix, as suggested in the link, you can download the channel by adding the channel to the download queue. This will pick up the entire list of videos and add them to the queue. If you only want specific videos from playlists, you can do the same there by adding those playlists to the video queue.

Let us know if you have any additional questions regarding this feature. As a note, you can modify the number of videos in the `Settings`, under `Applications Configurations` > `Subscriptions`. This number is how far back into the video list each scrape attempt will look - keeping this number low will decrease the likelihood of issues from YouTube's side, however, especially for larger channels.

**Comment by x775** [comment]
Thanks for your swift response and clarification, @lamusmaser.

It is reassuring to hear I am not the only one who has misunderstood this, though at the same time, perhaps that also suggests some clarification would be beneficial. Specifically, on the Settings-page, the description for YouTube's page size currently reads _"Videos to scan to find new items for the Rescan subscriptions task, max recommended 50."_  It is not at all clear to me that this also dictates the number of videos to download upon first adding a channel. Indeed, the explicit mention of "rescan" (and as per https://github.com/tubearchivist/tubearchivist/wiki/Downloads#rescan-subscriptions) makes it sound as though it is strictly the number of most recent videos that will be re-scanned to check for new uploads.

Similarly, under https://github.com/tubearchivist/tubearchivist/wiki/Downloads#add-to-download-queue, I see:

> Add a Channel ID or Channel URL to add every available video to the download queue. **This will ignore the channel page size as described before and is meant for an initial download of the whole channel.** You can still ignore selected videos before starting the download. 
(emphasis mine)

**Comment by lamusmaser** [comment]
Yep, it is something that has come up multiple times. I believe that there has been some discussion of rewording several areas to increase clarity, but some of those changes may be waiting for the new front-end that is being created.

**Comment by x775** [comment]
> Yep, it is something that has come up multiple times. I believe that there has been some discussion of rewording several areas to increase clarity, but some of those changes may be waiting for the new front-end that is being created.

Sounds good, thank you!

I hit enter prematurely in my above response, but have now finalised it.

**Comment by bbilly1** [comment]
This is one of these things that makes sense from a programming point of view. Basically a *rescan* of your subscriptions gets the latest x videos from youtube, looks if all are indexed, and adds any missing videos to the queue. That does the exact same thing if you haven't downloaded any videos from this channel or if you are just missing the newest video from that channel.

But I can see how this might not be obvious from a user point of view. But I hope with the FAQ entry, this has become clearer. When subscribing, you might not automatically want to download the whole channel, but you obviously can.

**Comment by Marthisdil** [comment]
> This is working as designed, but this particular issue has been brought up a few times.
> 
> From the Wiki's FAQ: https://github.com/tubearchivist/tubearchivist/wiki/FAQ#5-when-i-subscribe-to-a-channel-it-only-downloads-the-most-recent-50-videos
> 
> As a quick fix, as suggested in the link, you can download the channel by adding the channel to the download queue. This will pick up the entire list of videos and add them to the queue. If you only want specific videos from playlists, you can do the same there by adding those playlists to the video queue.
> 
> Let us know if you have any additional questions regarding this feature. As a note, you can modify the number of videos in the `Settings`, under `Applications Configurations` > `Subscriptions`. This number is how far back into the video list each scrape attempt will look - keeping this number low will decrease the likelihood of issues from YouTube's side, however, especially for larger channels.

As has also been posted here, and in Discord, TA doesn't *always* get all the videos when adding the /videos link to the download screen.

There's no other way to grab JUST the missing videos but to wait for the download to finish, and then put the URL in again and have TA rescan all of the videos...just for that *1* channel.
